scripts/sandbox.js

```diff
@@ -112,49 +114,71 @@ async function getChallenges() {
 }
 getChallenges();
 
-// get ratings
-
+//Set the rating stars on the challange cards
 function setStarRating() {
   let ratingsUl = document.querySelectorAll("ul.rating");
-  //console.log("ratingUl", ratingsUl);
   for (let j = 0; j < ratingsUl.length; j++) {
-    //console.log("ratingUl", ratingsUl[j]);
     let ratingChil = ratingsUl[j].children;
     let ratingsList = Array.from(ratingChil);
-    //console.log("ratingsChil", ratingChil);
-    //console.log("ratingsList", ratingsList);
 
-    let ratValue = Math.ceil(ratingsUl[j].getAttribute("aria-valuenow"));
-    //console.log(ratValue);
+    //the int part of rating value
+    let ratingValueInt = Math.floor(ratingsUl[j].getAttribute("aria-valuenow"));
+    //the decimal part of rating value
+    let ratingValueDecimal = (ratingsUl[j].getAttribute("aria-valuenow")%1).toFixed(1);
+
     for (let i = 0; i < ratingsList.length; i++) {
-      if (ratingsList == 0) {
+      //0-star rated challenge
+      if (ratingValueInt == 0) {
         let star = ratingsList[i];
         star.classList.remove("active");
-      } else if (ratValue == 1) {
+        //0.5 star rated challenge
+        if(i == 0 && ratingValueDecimal != 0){
+          star.classList.add("half");
+        }
+        //1 star rated challenge
+      } else if (ratingValueInt == 1) {
```

Comment:
The logic here is unnecessarily complex and it should be possible to simplify it using math instead of having one if condition (branch) for every value.